docs/source/apis/Products.rst

```diff
@@ -0,0 +1,412 @@
+Products
+########
+
+According to `Amazon's documentation
+<https://docs.developer.amazonservices.com/en_US/products/Products_Overview.html>`_:
+
+   The Products API section of Amazon Marketplace Web Service (Amazon MWS) helps you get information to match your
+   products to existing product listings on Amazon Marketplace websites and to make sourcing and pricing decisions for
+   listing those products on Amazon Marketplace websites. The Amazon MWS Products API returns product attributes,
+   current Marketplace pricing information, and a variety of other product and listing information.
+
+Using examples on this page
+===========================
+
+All examples below assume you have setup your Products API instance appropriately. Refer to :doc:`../gettingStarted`
+for details:
+
+.. code-block:: python
+
+    from mws import Products
+
+    products_api = Products(
+        access_key="...",
+        secret_key="...",
+        account_id="...",
+        auth_token="...",
+    )
+
+All request methods in the Products API also require a **MarketplaceId** to specify which marketplace the products
+are sold in. MarketplaceId values should match one of the values specified in Amazon documentation:
+`Amazon MWS endpoints and MarkeplaceId values <https://docs.developer.amazonservices.com/en_US/dev_guide/DG_Endpoints.html>`_
+
+python-amazon-mws makes these values available through the :py:class:`Marketplaces <mws.mws.Marketplaces>` Enum, which
+contains both the ``endpoint`` and ``marketplace_id`` for each Amazon region via that region's country code.
+
+For convenience, a ``Marketplaces`` instance will return its MarketplaceId through the ``.value`` attribute, as well.
+Further, all request methods in python-amazon-mws will automatically "clean" Enum instances by returning their ``.value``
+attributes.
+
+The following are all valid methods for obtaining, for example, the MarketplaceId for the US region and passing it
+to a request method in the ``Products`` API:
+
+.. code-block:: python
+
+    from mws import Marketplaces
+
+    my_market = Marketplaces.US
+    # Returns the Enum instance for the US region.
+    # When used in a request method, the `marketplace_id` value will be used automatically.
+
+    print(my_market.marketplace_id)
+    # 'ATVPDKIKX0DER'
+    print(my_market.value)
+    # 'ATVPDKIKX0DER'
+    # (alias for `.marketplace_id`)
+
+    # You can also return the endpoint for that region, if needed:
+    print(my_market.endpoint)
+    # 'https://mws.amazonservices.com'
+
+In all examples below, replace ``my_market`` with the ``Marketplaces`` Enum instance or MarketplaceId string value
+relevant to your region.
+
+Products API reference
+======================
+
+.. ################################################# Class definition ##################################################
+.. autoclass:: mws.apis.products.Products
```

Comment:
Here we see `autodoc` at work, automatically showing the class `__init__` method and other method signatures, without needing to rewrite them; as well as the docstrings from those methods.

To aid in developer readability, and because of the number of examples added to the `automethod` calls below here, I opted for a commented-out separator in the doc source code, using `.. ##### foo #####` all the way across the full width of the file (up to about 120 characters).